### DISCUSSION TITLE
Need help installing pytorch geometric properly

### DISCUSSION THREAD

**Discussion by ishanikarmarkar** [pyg-team/pytorch_geometric#9659] [Q&A]
Hello, I am trying to install pytorch geometric properly on an M3-Pro chip machine. After days of trying, I was able to install it with torch 2.3.0. import torch and import torch_geometric execute perfectly fine. However, when I try "from torch_geometric.datasets import TUDataset" I get the following error. 

Can someone please provide any assistance? I'm completely desperate, nothing seems to be working. 

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[3], line 9
      7 import torch.nn.functional as F
      8 from torch.nn import BatchNorm1d, Linear, ReLU, Sequential, BCEWithLogitsLoss
----> 9 from torch_geometric.datasets import TUDataset
     10 from torch_geometric.loader import DataLoader
     11 from torch_geometric.nn import GINConv, global_add_pool

File /opt/anaconda3/envs/pyg_env/lib/python3.9/site-packages/torch_geometric/__init__.py:13
     11 from .home import get_home_dir, set_home_dir
     12 from .device import is_mps_available, is_xpu_available, device
---> 13 from .isinstance import is_torch_instance
     14 from .debug import is_debug_enabled, debug, set_debug
     16 import torch_geometric.utils

File /opt/anaconda3/envs/pyg_env/lib/python3.9/site-packages/torch_geometric/isinstance.py:7
      3 import torch
      5 import torch_geometric.typing
----> 7 if torch_geometric.typing.WITH_PT20:
      8     import torch._dynamo
     11 def is_torch_instance(obj: Any, cls: Union[Type, Tuple[Type]]) -> bool:

AttributeError: partially initialized module 'torch_geometric' has no attribute 'typing' (most likely due to a circular import)

**Comment by akihironitta** [comment]
I cannot reproduce it on the latest release. Would you mind updating it to `2.6.1` and retrying it?

**Comment by ShadovvSinger** [comment]
I've observed this issue and the networkx issue at the same time (one after another).
by following https://github.com/Aider-AI/aider/issues/3171#issuecomment-3377853141, I achieved to stop observing both. It's worth trying this.
